micro-rdk/src/common/conn/server.rs

```diff
@@ -470,6 +474,16 @@ where
                     let _ = AsyncRwLockUpgradableReadGuard::upgrade(urguard)
                         .await
                         .insert(app_client);
+                } else {
+                    let current_count = urguard.as_ref().unwrap().get_grpc_client_count();
+                    log::info!(
+                        "ViamServer signaling flow has created grpc client clone {} of {}",
+                        current_count,
+                        max_anticipated_grpc_clients
+                    );
+                    if current_count > (max_anticipated_grpc_clients - 1) {
```

Comment:
nit: might be worth just repeating the `self.app_client_tasks.len()` here. don't have to think about what the -1 when looking at this line

Reply:
Or just write it as `= max...`. Or `>=` if you want to be a little paranoid.